# Práctica 1 - Text-Mining
<div style="border-top: 2px solid white; width: 550px; margin: 5px 0;"></div>

### Asignatura de Minería de Datos | Universidad Carlos III de Madrid
<div style="border: 2px solid white; padding: 10px; width: fit-content;">
Alonso Velázquez Torres<br>
Jorge López García<br>
Isaac Gil Vélez<br>
Carlos Balsa Gómez<br>
</div>

*AQUÍ PONER INTRODUCCIÓN A LA PRÁCTICA*

El primer paso, como en todo proyecto de programación, es importar todas las librerías que necesitaremos:

In [3]:
import pandas as pd

Realizaremos nuestra práctica de text mining sobre el dataset de NoticIA, por tanto, lo primero que debemos hacer es importarlo y a su vez para ello necesitaremos importar la función load_datasets del paquete datasets.

In [4]:
from datasets import load_dataset
ds = load_dataset('Iker/NoticIA')

e:\Universidad\AÑO 4\Mineria de datos\.venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Como se puede ver en el output del bloque de código anterior, el dataset ya viene dividido en índices de entrenamiento, validación y test.
Como no nos interesa entrenar ningún modelo en esta práctica, los concatenaremos en un sólo dataset para disponer de mayor cantidad de texto. Lo haremos según sigue:

In [5]:
df = pd.concat([
 ds['train'].to_pandas(),
 ds['validation'].to_pandas(),
 ds['test'].to_pandas()
 ], ignore_index=True)

Ahora que ya tenemos un único dataset, vamos a proceder a ejecutar algunas funciones de análisis exploratorio:

In [ ]:
print(df.size)
print(df.shape)
print(df.columns)
df.head()

3400
(850, 4)
Index(['web_url', 'web_headline', 'summary', 'web_text'], dtype='object')


web_url  \
0  https://www.informacion.es/medio-ambiente/2023...   
1                            https://buff.ly/3Rdqz2K   
2                      https://www.cope.es/n/2978369   
3                   https://www.genbeta.com/p/316184   
4                            https://buff.ly/47SBCap   

                                        web_headline  \
0  JORGE REY: EL TIEMPO | La impactante predicció...   
1  El cambio en las matrículas que se espera para...   
2  Si no avisas a la DGT de este cambio en tu coc...   
3  Estos serán los lenguajes de programación con ...   
4  Cambio de estrategia en Microsoft: Windows 12 ...   

                                             summary  \
0              El inicio de un periodo frío intenso.   
1                     Se dará el salto a la letra M.   
2  500 euros por pintar un coche de otro color y ...   
3                               Python y JavaScript.   
4  Solo un 28.6% de los usuarios actuales de Wind...   

                                            web_text  
0  27·11·23 | 08:34 | Actualizado a las 14:47\nJO...  
1  Si eres de los que sigues el avance de las mat...  
2  Con Pilar Cisneros y Fernando de Haro\nCon Pac...  
3  Si con el año nuevo te has propuesto aumentar ...  
4  Desde hace ya varios meses, las especulaciones...

Vemos que el dataset contiene 3400 elementos, ordenados en 850 entradas para cada una de las 4 columnas del dataset.
Las columnas hacen referencia a la url de cada noticia, su título, un resumen de su contenido y su contenido en sí.

También podemos identificar que los textos están en español, mezclan minúsculas con mayúsculas, y a veces incluyen letras u otros símbolos.

### Preprocesamiento del texto
Como estamos trabajando con un conjunto de datos no estructurado de tipo texto, no podemos aplicar otras técnicas de análisis exploratorio de datos como medias, frecuencias o correlaciones. 

En su lugar, tendremos que realizar lo que se conoce como preprocesamiento del texto.